pkg/spi-shared/config/config.go

```diff
@@ -57,7 +73,28 @@ type CommonCliArgs struct {
 // consumption.
 type persistedConfiguration struct {
 	// ServiceProviders is the list of configuration options for the individual service providers
-	ServiceProviders []ServiceProviderConfiguration `yaml:"serviceProviders"`
+	ServiceProviders []persistedServiceProviderConfiguration `yaml:"serviceProviders"`
+}
+
+// ServiceProviderConfiguration contains configuration for a single service provider configured with the SPI. This
+// mainly contains config.yaml of the OAuth application within the service provider.
+type persistedServiceProviderConfiguration struct {
+	// ClientId is the client ID of the OAuth application that the SPI uses to access the service provider.
+	ClientId string `yaml:"clientId"`
+
+	// ClientSecret is the client secret of the OAuth application that the SPI uses to access the service provider.
+	ClientSecret string `yaml:"clientSecret"`
+
+	// ServiceProviderType is the type of the service provider. This must be one of the supported values: GitHub, Quay
```

Comment:
the name in the doc comment no longer corresponds to the name of the field. Also, Gitlab is not mentioned as a possible value.

Reply:
fixed